Error installing - ../src/GNConstants.cpp:239:32: error: ‘GETDNS_RCODE_COOKIE’

built and installed getdns 
then npm install --save getdns

After a number of warnings about things being deprecated compilation fails with
`  CXX(target) Release/obj.target/getdns/src/GNConstants.o
../src/GNConstants.cpp: In static member function ‘static void GNConstants::Init(v8::Local<v8::Object>)’:
../src/GNConstants.cpp:239:32: error: ‘GETDNS_RCODE_COOKIE’ was not declared in this scope; did you mean ‘GETDNS_RCODE_BADCOOKIE’?
  239 |     SetConstant("RCODE_COOKIE",GETDNS_RCODE_COOKIE,exports);
      |                                ^~~~~~~~~~~~~~~~~~~
      |                                GETDNS_RCODE_BADCOOKIE
make: *** [getdns.target.mk:113: Release/obj.target/getdns/src/GNConstants.o] Error 1
make: Leaving directory '/home/francis/src/resolv/node_modules/getdns/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Linux 5.4.0-72-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/francis/src/resolv/node_modules/getdns
gyp ERR! node -v v14.16.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm WARN resolv@1.0.0 No description
npm WARN resolv@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! getdns@4.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the getdns@4.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/francis/.npm/_logs/2021-05-11T11_49_44_685Z-debug.log
`

--- a/src/GNConstants.cpp
+++ b/src/GNConstants.cpp
@@ -236,5 +236,19 @@ void GNConstants::Init(Local<Object> target) {
     SetConstant("RCODE_BADNAME",GETDNS_RCODE_BADNAME,exports);
     SetConstant("RCODE_BADALG",GETDNS_RCODE_BADALG,exports);
     SetConstant("RCODE_BADTRUNC",GETDNS_RCODE_BADTRUNC,exports);
+
+// NOTE: workaround for breaking change in upstream getdns versions after getdns v1.6.0; GETDNS_RCODE_COOKIE constant was renamed to GETDNS_RCODE_BADCOOKIE.
+// NOTE: the renamed constant has not yet been included in an upstream getdns released, so this is a preemptive measure.
+// https://github.com/getdnsapi/getdns-node/issues/38
+// https://github.com/getdnsapi/getdns/pull/471
+// https://github.com/getdnsapi/getdns/commit/5c79e2c7315f0aa235bb0b2adf26ce2129786e7f
+#ifdef GETDNS_RCODE_BADCOOKIE
+    SetConstant("RCODE_BADCOOKIE",GETDNS_RCODE_BADCOOKIE,exports);
+#else
+    // TODO: remove deprected legacy COOKIE constant support a few major getdns-node releases after BADCOOKIE has been made released in the upstream getdns.
     SetConstant("RCODE_COOKIE",GETDNS_RCODE_COOKIE,exports);
+
+    // NOTE: transitioning the constant; use BADCOOKIE instead of COOKIE (deprecated).
+    SetConstant("RCODE_BADCOOKIE",GETDNS_RCODE_COOKIE,exports);
+#endif
 }

